functions/controllers/user/login.js

```diff
@@ -0,0 +1,60 @@
+const firebase = require("firebase");
+const { validateLoginData } = require("../../util/validator");
+const { status, message } = require("../../util/constants");
+var HttpStatus = require("http-status-codes");
+
+const loginUser = async (req, res, db ) => {
+    const { email, password } = req.body;
+    const user = {
+      email,
+      password
+    };
+    const { error, success } = status;
+    const {
+      somethingWentWrong,
+      wrongCredentials,
+      auth_userNotFound,
+      auth_wrongPassword,
+    } = message;
+    const {
+      PRECONDITION_FAILED,
+      INTERNAL_SERVER_ERROR,
+      BAD_REQUEST,
+      OK,
+    } = HttpStatus;
+
+    const { errors, valid } = validateLoginData(user);
+    if (!valid)
+      return res
+        .status(PRECONDITION_FAILED)
+        .json({ status: error, message: errors });
+  
+    try {
+      const userData = await firebase
+        .auth()
+        .signInWithEmailAndPassword(user.email, user.password);
+      const Token = await userData.user.getIdToken();
+      return res.status(OK).json({ status: success, message: Token });
```

Comment:
![image](https://user-images.githubusercontent.com/2178492/69865346-39fae500-12a1-11ea-95a9-bc92b358e9c3.png)

Follow JS End convention for success message; use "data" instead of "message"